docs/DeveloperGuide.md

```diff
@@ -1072,6 +1072,199 @@ testers are expected to do more *exploratory* testing.
+1. Prerequisites: List all students using the `list` command. Multiple students in the list.
+
+2. Test case: `find c/CS2103T` (Assuming Students with course `CS2103T` Exist)<br>
+   Expected: Displays students details with course `CS2103T`.
+3. Test case: `find c/CS2103T` (Assuming Students with course `CS2103T` does not Exist)<br>
+   Expected: No Students Found. Displays 0 students.
+4. Test case: `find c/1234`
+   Expected: No Students Found. Error details shown.
+
+
+### Finding a student (by name)
+1. Prerequisites: List all students using the `list` command. Multiple students in the list.
+
+2. Test case: `find n/TestOne` (Assuming Student with name `TestOne` Exists)<br>
+   Expected: Displays students details with name `TestOne`.
+3. Test case: `find n/TestOne` (Assuming Students with name `TestOne` does not Exist)<br>
+   Expected: No Students Found. Displays 0 students.
+4. Test case: `find n/Test1`
+   Expected: No Students Found. Error details shown.
+
+### Editing a student
+1. Prerequisites: List all students using the `list` command. Multiple students in the list.
+
+2. Test case: `edit 1 n/TestOne p/11111111`<br>
+   Expected: 1st student is edited. Details of the edited student is shown.
+3. Test case: `edit 2 e/test1@example.com c/CS2103T`<br>
+   Expected: 2nd student is edited. Details of the edited student is shown.
+4. Test case: `edit 2 n/Test 2`<br>
+   Expected: No student is edited. Error details shown.
+5. Other incorrect edit commands to try: `edit`, `edit x`, `...` (where x is larger than the list size)<br>
+   Expected: No student is edited. Error details shown.
+
+### Deleting a student
+1. Prerequisites: List all students using the `list` command. Multiple students in the list.
+
+2. Test case: `delete 1`<br>
+   Expected: 1st student is deleted from the list. Details of the deleted student shown in the status message.
+3. Test case: `delete 0`<br>
+   Expected: No student is deleted. Error details shown in the status message.
+4. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+   Expected: No student is deleted. Error details shown in the status message.
+
+### Lesson Test Cases
+
+### Adding a lesson: `addlesson`
+
+1. Test case: `addlesson d/2024-10-20 t/14:00`  
+   Expected: Lesson with date 2024-10-20 and time 14:00 is added to the list. Details of the added lesson are shown.
+2. Test case: `addlesson d/2024-10-20`  
+   Expected: No lesson is added. Error details shown.
+3. Test case: `addlesson t/14:00`  
+   Expected: No lesson is added. Error details shown.
+4. Test case: `addlesson d/2024-10-20 t/14:00` (when a lesson with the same date and time already exists)  
+   Expected: No lesson is added. Error message about duplicate lesson shown.
+
+### Adding a student to a lesson: `addtolesson`
+
+1. Prerequisites: At least one lesson and one student exist in the list.
+
+2. Test case: `addtolesson 1 n/TestOne`  
```

Comment:
can add that u can add multiple students to a lesson in 1 command
e.g. `addtolesson 1 n/TestOne i/2`